packages/medusa/src/services/product.ts

```diff
@@ -49,6 +58,9 @@ class ProductService extends TransactionBaseService<ProductService> {
   protected readonly productTagRepository_: typeof ProductTagRepository
   protected readonly imageRepository_: typeof ImageRepository
   protected readonly productVariantService_: ProductVariantService
+  protected get salesChannelService_(): SalesChannelService {
+    return this.container.salesChannelService as SalesChannelService
```

Comment:
question: We have a circular deps, the quick fix is to use a getter in order to defer the access to the deps. We should think of something else or atm should we go that way?

Reply:
suggestion: I think it might be worth thinking of a more scalable way of getting rid of the circular deps, since I dread that this pattern could lead to architectural issues down the line.

Could we remove the `ProductService` deps from the `SalesChannelService` by moving the error handling of `SalesChannelService.addProducts` to live the repository level instead?

Reply:
I noticed, that the error handling performed in both services are almost identical -> Reacting to a foreign key constraint error and displaying an error to the end user. Could make sense to investigate a generic pattern for this, but I don't have an idea top of mind

Reply:
For now, we could go with the first mentioned approach. Wdyt? 

cc @srindom @fPolic @pKorsholm

Reply:
I don't think we can move the error handling to the repository since the error thrown will abort the transaction, thus we cannot continue querying. I believe this is the reasoning to add the error handler to the atomicphase initially.

We could add an additional repository method to the product repository which would find which of the sales channel id's are not actual sales channels, and that method could be invoked in the error handler instead. However, adding a method for dealing with sales channels in the product repository, doesn't sound like good design either.

I think deferring the loading of the service like @adrien2p has done is not the worst solution. We could potentially add some additional error handling in the get method in order to validate the container field since the `get` method allows us to introduce additional behavior and validation?

Reply:
> I don't think we can move the error handling to the repository since the error thrown will abort the transaction, thus we cannot continue querying. I believe this is the reasoning to add the error handler to the atomicphase initially.

My idea was to add a repository method to the SalesChannel repository responsible for adding products + throwing on non-existing ones which is caught by the atomic phase

Reply:
I quite like the base repo method approach, since I could imagine this exact case being recurring as a result of our new bulk add / remove pattern. What do you think @pKorsholm?

Reply:
yeah, that would be great! 😄 Love that

Reply:
ok so should I include it in that PR then?

Reply:
We might as well, if you don't mind 👍

Reply:
consider it done already